app/app.go

```diff
@@ -1121,9 +1129,43 @@ func (app *App) ProcessBlockConcurrent(
 	return txResults, ok
 }
 
+func (app *App) ProcessTxs(
+	ctx sdk.Context,
+	txs [][]byte,
+	dependencyDag *acltypes.Dag,
+	processBlockConcurrentFunction ProcessBlockConcurrentFunction,
```

Comment:
why do we pass the the function instead of directly calling the `ProcessBlockConcurrent`? is it primarily for the mocking purposes in testing? if so, isn't that an antipattern? (I don't have strong feelings one way or the other on whether we keep it this way, just looking to better understand this & golang best practices.

Reply:
Yeah, this is mainly for testing, I was following this: https://www.myhatchpad.com/insight/mocking-techniques-for-go/